wallet/wallet-rpc-lib/src/rpc/server_impl.rs

```diff
@@ -1104,16 +1029,16 @@ where
         token_id: Option<RpcAddress<TokenId>>,
         htlc: RpcHashedTimelockContract,
         options: TransactionOptions,
-    ) -> rpc::RpcResult<HexEncoded<SignedTransaction>> {
-        let config = ControllerConfig {
-            in_top_x_mb: options.in_top_x_mb(),
-            broadcast_to_mempool: true,
-        };
-
+    ) -> rpc::RpcResult<RpcNewTransaction> {
```

Comment:
Htlc tx is never broadcasted automatically. I wonder if we should have a separate type similar to NewTransaction that doesn't have the `broadcasted` field and instead implies that it's always false.
Also, we could re-use that type for `make_tx_for_sending_tokens_with_intent` and `make_tx_to_send_tokens_from_multisig_address`.

P.S. we do have the `ComposedTransaction` type, so I guess we could re-use it instead. But probably it should be made consistent with `RpcNewTransaction`. (i.e. contain the same fields except for `broadcasted`).

Reply:
Added a new type similar to NewTransaction without the broadcast